src/openssl/mod.rs

```diff
@@ -230,12 +230,11 @@ impl ECVRF {
     fn generate_nonce(&mut self, secret_key: &BigNum, data: &[u8]) -> Result<BigNum, Error> {
         // Bits to octets from data - bits2octets(h1)
         // Length of this value should be dependent on qlen (i.e. `SECP256k1` is 32)
-        //FIXME: VRF-draft-04 test vectors were computed with a wrong `qlen` parameter for `bits2octets`
-        let mod_qlen = match self.cipher_suite {
-            CipherSuite::P256_SHA256_TAI => data.len() * 8, // should be 32 instead of 33 bytes
-            _ => self.qlen,
-        };
-        let data_trunc = bits2octets(data, mod_qlen, &self.order, &mut self.bn_ctx)?;
+        // We follow the new VRF-draft-05 in which the input is hashed`
+        let data_hash = hash(self.hasher, &data)?;
+
+        let mod_qlen = self.qlen;
```

Comment:
Instead of assigning `mod_qlen` to be used in the next function, we could use directly `self.qlen`.